sdk/resourcemanager/Azure.ResourceManager/src/ArmClient.cs

```diff
@@ -42,50 +38,28 @@ protected ArmClient()
         {
         }
 
-        /// <summary>
-        /// Initializes a new instance of the <see cref="ArmClient"/> class.
-        /// </summary>
-        /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
-        /// <param name="options"> The client parameters to use in these operations. </param>
-        /// <exception cref="ArgumentNullException"> If <see cref="TokenCredential"/> is null. </exception>
-        public ArmClient(TokenCredential credential, ArmClientOptions options = default) : this(credential, null, _defaultUri, options)
-        {
-        }
-
         /// <summary>
         /// Initializes a new instance of the <see cref="ArmClient"/> class.
         /// </summary>
         /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
         /// <param name="defaultSubscriptionId"> The id of the default Azure subscription. </param>
         /// <param name="options"> The client parameters to use in these operations. </param>
         /// <exception cref="ArgumentNullException"> If <see cref="TokenCredential"/> is null. </exception>
-        public ArmClient(TokenCredential credential, string defaultSubscriptionId, ArmClientOptions options = default) : this(credential, defaultSubscriptionId, _defaultUri, options)
-        {
-        }
-
-        /// <summary>
-        /// Initializes a new instance of the <see cref="ArmClient"/> class.
-        /// </summary>
-        /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
-        /// <param name="defaultSubscriptionId"> The id of the default Azure subscription. </param>
-        /// <param name="baseUri"> The base URI of the service. </param>
-        /// <param name="options"> The client parameters to use in these operations. </param>
-        /// <exception cref="ArgumentNullException"> If <see cref="TokenCredential"/> is null. </exception>
-        public ArmClient(TokenCredential credential, string defaultSubscriptionId, Uri baseUri, ArmClientOptions options = default)
+        public ArmClient(TokenCredential credential, string defaultSubscriptionId = default, ArmClientOptions options = default)
```

Comment:
This is the most concise way to code the parameter overloads, but the feedback from customers / archboard is that we want to have non optional ctors.  We should have exactly these 3 overloads here.

ArmClient(TokenCredential credential)
ArmClient(TokenCredential credential, string defaultSubscriptionId)
ArmClient(TokenCredential credential, string defaultSubscriptionId, ArmClientOptions options)

Reply:
done